src/main/java/com/segment/analytics/android/integrations/optimizelyx/OptimizelyXIntegration.java

```diff
@@ -1,63 +1,102 @@
  *
- * @see <a href="https://www.optimizely.com/">Pptimizely</a>
- * @see <a href="https://segment.com/docs/integrations/optimizely/">Optimizely Integration</a>
- * @see <a href="http://developers.optimizely.com/android/">Optimizely Android SDK</a>
+ * @see <a href="https://www.optimizely.com/">Optimizely</a>
+ * @see <a href="https://segment.com/docs/destinations/optimizelyx/">Optimizely Integration</a>
+ * @see <a
+ *     href="https://developers.optimizely.com/x/solutions/sdks/introduction/index.html?language=android&platform=mobile">Optimizely
+ *     Android SDK</a>
  */
-public class OptimizelyXIntegration extends Integration<Void> {
-  private static final String OPTIMIZELY_KEY = "Optimizely";
-
-  private final Context context;
-  private final Logger logger;
-  private final Analytics analytics;
-
-  // Optimizely must be initialized immediately on launch. So we require the token when creating the
-  // factory, and initialize it in this method.
-//  public static Factory createFactory(final Application application, String token) {
-////    Optimizely.startOptimizelyWithAPIToken(token, application);
-////    return new Factory() {
-////      @Override public Integration<?> create(ValueMap settings, Analytics analytics) {
-////        boolean listen = settings.getBoolean("listen", false);
-////        Logger logger = analytics.logger(OPTIMIZELY_KEY);
-////
-////        OptimizelyIntegration integration =
-////            new OptimizelyIntegration(analytics, application, logger);
-////        if (listen) {
-////          Optimizely.addOptimizelyEventListener(integration);
-////        }
-////        return integration;
-////      }
-//
-////      @Override public String key() {
-////      return OPTIMIZELY_KEY;
-//////      }
-//    };
-//  }
-
-  public OptimizelyXIntegration(Analytics analytics, Context context, Logger logger) {
-    this.analytics = analytics;
-    this.context = context;
-    this.logger = logger;
-  }
+public class OptimizelyXIntegration extends Integration<NotificationListener> {
+  static final Options options = new Options().setIntegration("Optimizely X", false);
+  private static final String OPTIMIZELYX_KEY = "Optimizely X";
+  final NotificationListener listener;
+
+  public static final Factory FACTORY =
+      new Factory() {
+        @Override
+        public Integration<?> create(ValueMap settings, Analytics analytics) {
+          return new OptimizelyXIntegration(analytics);
+        }
+
+        @Override
+        public String key() {
+          return OPTIMIZELYX_KEY;
+        }
+      };
+
+  OptimizelyXIntegration(final Analytics analytics) {
+
+    listener =
+        new NotificationListener() {
+          @Override
+          public void onEventTracked(
+              String eventKey,
+              String userId,
+              Map<String, String> attributes,
+              Long eventValue,
+              LogEvent logEvent) {
+
+            if (!isNullOrEmpty(attributes)) {
```

Comment:
Just so I understand, you calling `identify` with the Optimizely user attributes, correct?

Reply:
Yes! But dovetailing with your comment above, I agree that all we really care about is tracking when an experiment was activated. With this in mind, I think we should just remove our call to identify with Optimizely attributes, since users will prob be calling identify regardless.

Reply:
Jaime at Optimizely mentions that our mutual clients would like the ability to send attributes/traits of users to Optimizely so that Optimizely can use those attributes/traits for targeting. However, we can't really do much here because Optimizely X doesn't have a way to set these user attributes outside of an activate experiment event.

Reply:
I'm not sure if it would be helpful but we could set a user property for the given user with the variation and experiment viewed

Reply:
Hmmm that might be tricky, because a single user can be associated with many different experiments - I think this may result in trait bloat in other downstream tools and warehouses.